src/Razor/src/Microsoft.NET.Sdk.Razor/ValidateStaticWebAssetsUniquePaths.cs

```diff
@@ -0,0 +1,84 @@
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class ValidateStaticWebAssetsUniquePaths : Task
+    {
+        private const string BasePath = "BasePath";
+        private const string RelativePath = "RelativePath";
+        private const string TargetPath = "TargetPath";
+
+        [Required]
+        public ITaskItem[] StaticWebAssets { get; set; }
+
+        [Required]
+        public ITaskItem[] WebRootFiles { get; set; }
+
+        public override bool Execute()
+        {
+            var assetsByWebRootPaths = new Dictionary<string, ITaskItem>(StringComparer.OrdinalIgnoreCase);
+            for (var i = 0; i < StaticWebAssets.Length; i++)
+            {
+                var contentRootDefinition = StaticWebAssets[i];
+                if (!EnsureRequiredMetadata(contentRootDefinition, BasePath) ||
+                    !EnsureRequiredMetadata(contentRootDefinition, RelativePath))
+                {
+                    return false;
+                }
+                else
+                {
+                    var webRootPath = GetWebRootPath(
+                        contentRootDefinition.GetMetadata(BasePath),
+                        contentRootDefinition.GetMetadata(RelativePath));
+
+                    if (assetsByWebRootPaths.TryGetValue(webRootPath, out var existingWebRootPath) &&
+                        !string.Equals(contentRootDefinition.ItemSpec, existingWebRootPath.ItemSpec, StringComparison.OrdinalIgnoreCase))
+                    {
+                        Log.LogError($"Conflicting assets with the same path '{webRootPath}' for content root paths '{contentRootDefinition.ItemSpec}' and '{existingWebRootPath.ItemSpec}'.");
+                        return false;
+                    }
+                    else if(existingWebRootPath == null)
+                    {
+                        assetsByWebRootPaths.Add(webRootPath, contentRootDefinition);
+                    }
+                }
+            }
+
+            for (var i = 0; i < WebRootFiles.Length; i++)
+            {
+                var webRootFile = WebRootFiles[i];
+                var relativePath = webRootFile.GetMetadata(TargetPath);
+                var webRootFileWebRootPath = GetWebRootPath("/", relativePath);
+                if (assetsByWebRootPaths.TryGetValue(webRootFileWebRootPath, out var existingAsset))
+                {
+                    Log.LogError($"The static web asset '{existingAsset.ItemSpec}' has a conflicting web root path '{webRootFileWebRootPath}' with the project file '{webRootFile.ItemSpec}'.");
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        // Normalizes /base/relative \base\relative\ base\relative and so on to /base/relative
+        private string GetWebRootPath(string basePath, string relativePath) => $"/{Path.Combine(basePath, relativePath.TrimStart('.').TrimStart('/')).Replace("\\", "/").Trim('/')}";
```

Comment:
What's up with the `.` being trimmed from the start of the line? Why does that show up?

Reply:
Doesn't this produce different outputs if the files are `/dir1/file1.txt` and `\dir1\file1.txt`? Is that a problem?

Reply:
Nope, the webroot path always refers to where the file will end up on disk (relative to the wwwroot folder) so we want to use a canonical version of it that is the one defined by this part. Trimming the "." ensures that we combine the path properly no matter what the relative path format was.